gcp_variant_transforms/vcf_to_bq_common.py

```diff
@@ -115,3 +118,19 @@ def write_headers(merged_header, file_path):
   """Writes a PCollection of ``VcfHeader`` to location ``file_path``."""
   _ = (merged_header | 'WriteHeaders' >>
        vcf_header_io.WriteVcfHeaders(file_path))
+
+
+def _raise_error_on_unrecognized_flags(pipeline_args):
+  # type: (List[str]) -> None
+  """Raises an error if there are unrecognized flags."""
+  options = pipeline_options.PipelineOptions(pipeline_args).get_all_options()
+  for flag in _get_flag_names(pipeline_args):
+    if not any(str.startswith(option, flag) for option in options):
+      raise ValueError('The flag {} is unrecognized.'.format(flag))
+
+
+def _get_flag_names(pipeline_args):
+  # type: (List[str]) -> List[str]
+  """Returns a list of flag names that starts with `COMMAND_LINE_ARG_PREFIX`."""
+  return [arg[len(_COMMAND_LINE_ARG_PREFIX):] for arg in pipeline_args
```

Comment:
nit: I think it's easier if you just do `arg.strip('-')` so that it covers both `-f` and `--flag` (none of the existing beam options have a single dash, but they could). It would work as long as you don't have a flag ending with a `-`, but I think that's ok.

Reply:
ah, actually, there's an `lstrip('-')` command that does exactly that :)

Reply:
Cool! Thanks!